packages/react-debug-tools/src/ReactDebugHooks.js

```diff
@@ -175,6 +175,13 @@ function useCacheRefresh(): () => void {
   return () => {};
 }
 
+function useMemoCache(size: number): Array<any> {
+  const hook = nextHook();
+  const value = hook !== null ? hook.memoizedState[0] : new Array(size);
+  hookLog.push({primitive: 'MemoCache', stackError: new Error(), value});
+  return value;
```

Comment:
i'm not sure about this, i adapted from useMemo

Reply:
Not sure it even needs to appear in the DevTools, for now at least, given that it's not a hook in the traditional sense, but rather an implementation detail of the React runtime.